atlassian_jwt_auth/frameworks/common/decorators.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+from jwt.exceptions import InvalidIssuerError, InvalidTokenError
+from .utils import process_asap_token, _validate_claims
+
+
+def _with_asap(func=None, backend=None, issuers=None, required=None,
+               subject_should_match_issuer=None):
+    if backend is None:
+        raise ValueError(
+            'Invalid value for backend. Use a subclass instead.'
+        )
+
+    def with_asap_decorator(func):
+        @wraps(func)
+        def with_asap_wrapper(*args, **kwargs):
```

Comment:
Why not request a `request` here as an argument?

Reply:
Django passes the `request` to the view as an `arg`; not a `kwarg` and Flask doesn't pass it at all instead preferring the global request object that is automatically set.